troubleshooting/known_issues/SBPS_boot_fail.md

```diff
@@ -0,0 +1,114 @@
+
+```bash
+ncn-w002:~ # systemctl status sbps-marshal.service
+× sbps-marshal.service - System service that manages Squashfs images projected via iSCSI for IMS, PE, and other ancillary images similar to PE.
+     Loaded: loaded (/usr/lib/systemd/system/sbps-marshal.service; enabled; preset: disabled)
+     Active: failed (Result: exit-code) since Tue 2024-11-12 19:20:05 UTC; 18h ago
+   Duration: 2.232s
+    Process: 1239406 ExecStart=/usr/lib/sbps-marshal/bin/sbps-marshal (code=exited, status=1/FAILURE)
+   Main PID: 1239406 (code=exited, status=1/FAILURE)
+        CPU: 1.394s
+
+Nov 12 19:20:04 ncn-w002 sbps-marshal[1239406]:   File "/usr/lib/sbps-marshal/bin/sbps-marshal", line 8, in <module>
+Nov 12 19:20:04 ncn-w002 sbps-marshal[1239406]:     sys.exit(main())
+Nov 12 19:20:04 ncn-w002 sbps-marshal[1239406]:              ^^^^^^
+Nov 12 19:20:04 ncn-w002 sbps-marshal[1239406]:   File "/usr/lib/sbps-marshal/lib/python3.12/site-packages/bin/agent.py", line 270, in main
+Nov 12 19:20:04 ncn-w002 sbps-marshal[1239406]:     rootfs_s3_etag = artifact["link"]["etag"]
+Nov 12 19:20:04 ncn-w002 sbps-marshal[1239406]:                      ~~~~~~~~~~~~~~~~^^^^^^^^
+Nov 12 19:20:04 ncn-w002 sbps-marshal[1239406]: KeyError: 'etag'
+Nov 12 19:20:05 ncn-w002 systemd[1]: sbps-marshal.service: Main process exited, code=exited, status=1/FAILURE
+Nov 12 19:20:05 ncn-w002 systemd[1]: sbps-marshal.service: Failed with result 'exit-code'.
+Nov 12 19:20:05 ncn-w002 systemd[1]: sbps-marshal.service: Consumed 1.394s CPU time.
+```
+
+The complete log for the SBPS Marshal agent `systemd` service can be obtained from `journalctl` log as below:
+
+```bash
+    # journalctl -u sbps-marshal.service
+```
+
+## Workaround Description
+
+Step 1: Check whether the image to be projected has `etag` in the image manifest
+
+example #1:
+
+```bash
+# cray ims images describe 3f89b76d-7578-4ac3-8891-0f8b4335b06d
+arch = "x86_64" 
+created = "2024-08-15T14:38:35.897457+00:00" 
+id = "3f89b76d-7578-4ac3-8891-0f8b4335b06d" 
+name = "kpreslan-20240815093713" 
+
+[link] etag = ""  <---------------------- `etag` set to NULL
+path = "s3://boot-images/3f89b76d-7578-4ac3-8891-0f8b4335b06d/manifest.json" 
+type = "s3" 
+
+[metadata]
+```
+
+Or
+
+example #2:
+
+```bash
+# cray artifacts get boot-images "4715ca23-0203-4664-8d78-455f56cab922/manifest.json" json
+
+ncn-m001:~ # cat json
+{
+   "artifacts" : [
+      {
+         "link" : {
+            "path" : "s3://boot-images/4715ca23-0203-4664-8d78-455f56cab922/compute.squashfs",  <-------- No `etag` field
+            "type" : "s3"
+         },
+         "md5" : "b4598a2fd2674c33ba885eab4836dfc8",
+         "type" : "application/vnd.cray.image.rootfs.squashfs"
+      }
+   ],
+   "created" : "20240202221449",
+   "version" : "1.0"
+}
+```
+
+With the above examples given:
+ example #1 has the `etag` field set to NULL
+ example #2 has `etag` field missing.
+
+Step 2: Delete the image which has `etag` missing or set to NULL

```

Comment:
Is another possible solution to update the IMS record so that it does contain an etag field? Even a false etag might allow things to work.

Reply:
It's not the IMS record itself that is the problem, it's parsing the `manifest.json` that the IMS record points to that causes the problem when the `etag` doesn't exist in the `manifest.json` file.

Reply:
Thanks for the clarification. Regardless, I think the solution should be to fix the etag with a satisfactory value -- whatever that may be -- rather than deleting the images. People uploaded the images for a reason, and they're not going to want to delete them.